File: route/api_w_raw.py
from .tool.func import *

def api_w_raw(name = 'Test', rev = '', exist_check = ''):
    with get_db_connect() as conn:
        curs = conn.cursor()

        if exist_check != '':
            curs.execute(db_change("select title from data where title = ?"), [name])
            if data:
                return flask.jsonify({ 'exist' : '1' })
            else:
                return flask.jsonify({})
        else:
            if rev != '':
                curs.execute(db_change("select data from history where title = ? and id = ?"), [name, rev])
            else:
                curs.execute(db_change("select data from data where title = ?"), [name])

            data = curs.fetchall()
            if data:
                return flask.jsonify({
                    "title" : name, 
                    "data" : render_set(
                        doc_name = name, 
                        doc_data = data[0][0],
                        data_type = 'raw'
                    )
                })
            else:
                return flask.jsonify({})

File: route/api_w_raw.go
package route

import (
	"opennamu/route/tool"
)

func Api_w_raw(config tool.Config, doc_name string, exist_check string, rev string) map[string]any {
    db := tool.DB_connect()
    defer tool.DB_close(db)

    new_data := make(map[string]any)

    if !tool.Check_acl(db, doc_name, "", "render", config.IP) {
        new_data["response"] = "require auth"
    } else if exist_check != "" {
        title := ""
        exist := tool.QueryRow_DB(
            db,
            "select title from data where title = ? collate nocase",
            []any{ &title },
            doc_name,
        )

        if !exist {
            new_data["exist"] = false
        } else {
            new_data["exist"] = true
        }

        new_data["data"] = title
        new_data["response"] = "ok"
    } else {        
        exist := false

        data := ""
        hide := ""
        if rev != "" {
            exist = tool.QueryRow_DB(
                db,
                "select data, hide from history where title = ? and id = ?",
                []any{ &data, &hide },
                doc_name, rev,
            )
        } else {
            exist = tool.QueryRow_DB(
                db,
                "select data from data where title = ?",
                []any{ &data },
                doc_name,
            )
        }

        if !exist {
            new_data["response"] = "not exist"
        } else {
            check_pass := false
            if hide != "" {
                if tool.Check_acl(db, "", "", "hidel_auth", config.IP) {
                    check_pass = true
                } else {
                    new_data["response"] = "require auth"
                }
            } else {
                check_pass = true
            }

            if check_pass {
                new_data["title"] = doc_name
                new_data["data"] = data

                new_data["response"] = "ok"
            }
        }
    }

    return new_data
}
